src/inspected-window-helpers/overlay-helpers.js

```diff
@@ -74,35 +66,49 @@ export function setupOverlayHelpers() {
     } else {
       backgroundColor = getColorFromString(appName);
     }
-    div.style.background = backgroundColor;
 
-    const childDiv = div.appendChild(document.createElement("div"));
-    childDiv.style.display = "flex";
-    childDiv.style.flexDirection = node.clientHeight > 80 ? "column" : "row";
-    childDiv.style.alignItems = "center";
-    childDiv.style.justifyContent = "center";
-    childDiv.style.color =
-      options.color || options.textColor || getColorFromString(appName, 1);
-    childDiv.style.fontWeight = "bold";
-    childDiv.style.height = "100%";
-    childDiv.style.fontSize = "32px";
-
-    const appNameDiv = document.createElement("div");
-    appNameDiv.appendChild(document.createTextNode(appName));
-    childDiv.appendChild(appNameDiv);
-
-    if (options.textBlocks && options.textBlocks.length >= 1) {
-      options.textBlocks.forEach(textBlock => {
-        const textBlockDiv = document.createElement("div");
-        textBlockDiv.appendChild(document.createTextNode(textBlock));
-        childDiv.appendChild(textBlockDiv);
-      });
-    }
-
-    return div;
+    const { height, left, top, width } = node.getBoundingClientRect();
+    const domStr = `
+      <div
+        class="${className}"
+        style="
+          background: ${backgroundColor};
+          height: ${options.height || height + "px"};
+          left: ${options.left || left + "px"};
+          pointer-events: none;
+          position: ${options.position || "absolute"};
+          top: ${options.top || top + "px"};
+          width: ${options.width || width + "px"};
+          z-index: ${options.zIndex || 40};
+        "
+      >
+        <div style="
+          align-items: center;
+          color: ${options.color ||
+            options.textColor ||
+            getColorFromString(appName, 1)};
+          display: flex;
+          flex-direction: ${node.clientHeight > 80 ? "column" : "row"};
+          font-size: 2rem;
+          font-weight: bold;
+          height: 100%;
+          justify-content: center;
+        ">
+          <div>${appName}</div>
+          ${
+            options.textBlocks && options.textBlocks.length >= 1
+              ? options.textBlocks.map(textBlock => `<div>${textBlock}</div>`)
+              : ""
+          }
+        </div>
+      </div>
+    `;
+    const overlayEl = new DOMParser().parseFromString(domStr, "text/html").body
+      .firstChild;
```

Comment:
I like how easy it is to read. Are there any other benefits to the string that we then parse vs creating objects in memory?

Reply:
I don't know if we need to be too worried about perf. I was just curious if there were other benefits besides the awesome legibility.